toolchains/vivado.py

```diff
@@ -329,3 +331,25 @@ def add_runtimes(self):
 
     def versions(self):
         return {'yosys': self.yosys_ver(), 'vivado': self.vivado_ver()}
+
+
+class VivadoYosysUhdm(VivadoYosys):
+    '''Vivado PnR + Yosys synthesis using uhdm frontend'''
+    def __init__(self, rootdir):
+        VivadoYosys.__init__(self, rootdir)
+        self.synthtool = 'yosys'
+        self.synthoptions = [
+            '-iopad', '-arch xc7', '-flatten', 'frontend=surelog'
+        ]
+        self.toolchain = 'yosys-vivado-uhdm'
+        uhdm_yosys_path = shutil.which("uhdm-yosys")
+        # by default assume, uhdm-yosys is installed in symbiflow env
+        if uhdm_yosys_path is None:
+            uhdm_yosys_path = f"{self.rootdir}/env/conda/envs/symbiflow-env/bin/uhdm-yosys"
```

Comment:
I think we should error out if the path is not there, as uhdm_yosys cannot be found, unless there is a strong reason to have this default assumption.